src/extension.ts

```diff
@@ -407,6 +408,53 @@ function installCustomRequestHandlers(context: VSCode.ExtensionContext) {
   languageClient.onRequest(protocol.ShowTaintVulnerabilityRequest.type, showAllLocations);
 }
 
+async function isFileIgnoredForFolder(workspaceFolderPath: string, gitCommand: string): Promise<boolean> {
+  const {sout, serr} = await new Promise<{ sout: string, serr: string }>((resolve, reject) => {
+    ChildProcess.exec(
+        gitCommand,
+        {cwd: workspaceFolderPath},
+        (error: Error & { code?: 0 | 1 | 128 }, stdout, stderr) => {
+          if (error && (error.code !== 0 && error.code !== 1)) {
+            logToSonarLintOutput('Error on git gitCommand: ' + error);
+            reject(error);
+            return;
+          }
+          resolve({sout: stdout, serr: stderr});
+        }
+    );
+  });
+
+  if (serr) {
+    throw new Error(serr);
+  }
+
+  return Promise.resolve(sout.length > 0);
+}
+
+async function isIgnoredByScm(fileUri: string): Promise<boolean> {
+  const uriRelative = VSCode.workspace.asRelativePath(fileUri, false);
+  const command = `git check-ignore ${uriRelative}`;
+  let ignoredForAtLeastOneWorkspaceFolder = false;
+  const workspaceFolders = VSCode.workspace.workspaceFolders;
+  if (!isIterable(workspaceFolders)) {
+    return false;
+  }
+  for (const workspaceFolder of workspaceFolders) {
```

Comment:
To me the algorithm is wrong, and can easily consider a file as ignored while it is not. Say you have 2 workspace folders: folderA and folderB.
folderA/foo.js is ignored
folderB/foo.js is not ignored
When trying to analyze folderB/foo.js, uriRelative will be equal to foo.js. Then you will check for each workspace folder if foo.js is ignored, which is true for folderB. So you'll skip the analysis...
It would be better to first decode from which workspace folder the file belongs to, and then only check the ignore in this folder.